src/modelo.jl

```diff
@@ -0,0 +1,159 @@
+
+lista_professores = DataFrame(XLSX.readtable("preferenciasFinais.xlsx", "preferenciasFinais")...)
+
+lista_restricoes = DataFrame(XLSX.readtable("RestricoesFinais.xlsx", "RestricoesFinais")...)
+
+function FacultyAssign( DataFrameMaterias ,
+                        DataFramePreferencias,
+                        DataFrameRestricoes;
+                        ListaProfesPos = [6, 49, 4, 0, 38, 22, 19, 44, 45, 40, 0, 27, 0],
+                        ListaMateriasPos = 93:105,
+                        ListaProfesComCargos = [52, 20, 30, 46],
+                        ListaProfesSubs = [42, 53, 54],
+                        NumeroHorarios = 8,
+                        NumeroDias = 5,
+                        CargaHorariaComumMin = 8,
+                        CargaHorariaComumMax = 12,
+                        CargaHorariaCargoMin = 4,
+                        CargaHorariaCargoMax = 6,
+                        CargaHorariaSubsMin = 12,
+                        CargaHorariaSubsMax = 16
+                        )
+
+
+    P = size(DataFramePreferencias, 1) # número de professores e matérias
+    M = size(DataFramePreferencias, 2) - 1 # número de professores e matérias
+    H = NumeroHorarios # horarios 
+    D = NumeroDias # dias
+
+    # Gerando matriz de duração de aulas 
+    DU = zeros(D, H, M)
+    for k = 1:M
+        DU[:, :, k] .= DataFrameMaterias[1 + (k - 1) * 5: 5 + (k - 1) * 5, 2:(H + 1)]
+    end
+
+    # Gerando a matriz Binária de aula
+    HT = copy(DU) / 2
+
+    # Gerando a matriz das preferências
+    preferencias = Matrix(DataFramePreferencias[:, 2:(M+1)])
+
+    # Gerando a matriz de restrições 
+    restricoes = zeros(D, H, P)
+    for k = 1:P
+        restricoes[:, :, k] .= DataFrameRestricoes[1 + (k - 1) * 5: 5 + (k - 1) * 5, 2: end]
+    end
+
+
+    ## Completando professores da pos 
+    sorteioProfePos = sample(setdiff(1:P, ListaProfesPos), 3, replace = false)
+    ListaProfesPos[4] = sorteioProfePos[1]
+    ListaProfesPos[11] = sorteioProfePos[2]
+    ListaProfesPos[13] = sorteioProfePos[3]
+
+    preferencias[sorteioProfePos[1],96] = 5
+    preferencias[sorteioProfePos[2],103] = 5
+    preferencias[sorteioProfePos[3],105] = 5
+
+
+    # Definindo o modelo
+    model = Model(Gurobi.Optimizer)
+    @variable(model, x[1:P, 1:M] ≥ 0, Bin) #P*M var de decisão
+
+    @constraint(model,[p=1:P, d=1:D, h=1:H], sum(HT[d, h, t] * x[p,t] for t=1:M) ≤ 1)
+
+    @constraint(model, [t=1:M], sum(x[p,t] for p=1:P) == 1)
+
+    # Professores "comuns"
+    @constraint(model, [p in setdiff(1:P, hcat(ListaProfesComCargos', ListaProfesSubs'))], sum(DU[d, h, t]*x[p,t] for t=1:M, d=1:D, h=1:H) ≥ CargaHorariaComumMin)
+    @constraint(model, [p in setdiff(1:P, hcat(ListaProfesComCargos', ListaProfesSubs'))], sum(DU[d, h, t]*x[p,t] for t=1:M, d=1:D, h=1:H) ≤ CargaHorariaComumMax)
+
+    # Professores com cargos 
+    @constraint(model, [p in ListaProfesComCargos'], sum(DU[d, h, t]*x[p,t] for t=1:M, d=1:D, h=1:H) ≥ CargaHorariaCargoMin)
```

Comment:
```suggestion
    @constraint(model, [p in ListaProfesComCargos], sum(DU[d, h, t]*x[p,t] for t=1:M, d=1:D, h=1:H) ≥ CargaHorariaCargoMin)
```
Não é pra ser necessário esse `'`.